### PR TITLE
include: Remove compat include headers

### PR DESCRIPTION
**Pull request by galak**
Remove the compat headers as its been 2 releases since we introduced
them.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
hmm, looks like the deprecation warnings were not working correctly?

**Review by jhedberg — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by andyross — APPROVED** [review]

**Review by pabigot — COMMENTED** [review]
I guess this is technically ok, but it would be more "polite" to have a release where the default is disabled, so people get some warning about the change before they're forced to rework all their source.

**Review by dcpleung — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ manifest:
       revision: 6835bfc741bf15e98fb7971293913f770df6081f
       path: modules/hal/esp-idf
     - name: fatfs
-      revision: df96914dd989907f3a5de4cb95b116c2f305820d
+      revision: pull/2/head
```
Comment (ulfalizer) [inline west.yml]: Are these leftovers?

Comment (ulfalizer) [reply west.yml]: Nm... guess it should be DNM until those are in at least.

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**